#! /bin/sh /usr/share/dpatch/dpatch-run
## gmond-fix-compile-error.dpatch by Sebastian Harl <sh@tokkee.org>
##
## DP: gmond plugin: Fixed a typo.
## DP:
## DP: For some reason, the value_t members "derive" and "absolute" had been
## DP: spelled in all capital letters, obviously resulting in a build error.

@DPATCH@

diff a/src/gmond.c b/src/gmond.c
--- a/src/gmond.c
+++ b/src/gmond.c
@@ -542,9 +542,9 @@ static int staging_entry_update (const char *host, const char *name, /* {{{ */
   else if (ds_type == DS_TYPE_GAUGE)
     se->vl.values[ds_index].gauge = value.gauge;
   else if (ds_type == DS_TYPE_DERIVE)
-    se->vl.values[ds_index].DERIVE += value.derive;
+    se->vl.values[ds_index].derive += value.derive;
   else if (ds_type == DS_TYPE_ABSOLUTE)
-    se->vl.values[ds_index].ABSOLUTE = value.absolute;
+    se->vl.values[ds_index].absolute = value.absolute;
 
   se->flags |= (0x01 << ds_index);
 
